pkg/devfile/devfile.go

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright 2021 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package devfile
+
+import (
+	"github.com/devfile/api/v2/pkg/attributes"
+	"github.com/devfile/api/v2/pkg/devfile"
+	devfilePkg "github.com/devfile/library/pkg/devfile"
+	parser "github.com/devfile/library/pkg/devfile/parser"
+	data "github.com/devfile/library/pkg/devfile/parser/data"
+
+	appstudiov1alpha1 "github.com/redhat-appstudio/application-service/api/v1alpha1"
+)
+
+func ParseDevfileModel(devfileModel string) (data.DevfileData, error) {
+	// Retrieve the devfile from the body of the resource
+	devfileBytes := []byte(devfileModel)
+	parserArgs := parser.ParserArgs{
+		Data: devfileBytes,
+	}
+	devfileObj, _, err := devfilePkg.ParseDevfileAndValidate(parserArgs)
+	return devfileObj.Data, err
+}
+
+// ConvertHASApplicationToDevfile takes in a given HASApplication CR and converts it to
+// a devfile object
+func ConvertHASApplicationToDevfile(hasApp appstudiov1alpha1.HASApplication, gitOpsRepo string, appModelRepo string) (data.DevfileData, error) {
+	devfileVersion := string(data.APISchemaVersion220)
+	devfileData, err := data.NewDevfileData(devfileVersion)
```

Comment:
2Qs here

1. I guess we didnt discuss this, do we just resort to the latest devfile version always? devfile/library supports 2.2.0 because we want to deal with devfile/api master and put those related changes but since 2.2.0 is not public, lets keep it at 2.1.0
2. err not checked, do we just not worry about the err here since we used a valid version?

Reply:
1. I'll switch back to 2.1.0, we're not using anything in 2.2.0 AFAICT

2. Good catch, I'll fix that.

Reply:
Updated.